mutation.hh

```diff
@@ -82,8 +82,11 @@ public:
         : _ptr(std::make_unique<data>(std::move(schema), std::move(key), std::move(mp)))
     { }
     mutation(const mutation& m)
-        : _ptr(std::make_unique<data>(schema_ptr(m.schema()), dht::decorated_key(m.decorated_key()), m.partition()))
-    { }
+    {
+        if (m._ptr) {
+            _ptr = std::make_unique<data>(schema_ptr(m.schema()), dht::decorated_key(m.decorated_key()), m.partition());
+        }
```

Comment:
This is because `optimized_optional<>` which is an intrusive optional, reusing an already existing `set` bit in the object to store whether the value is set or not (assuming the value can also be set or not set already).
In the case of the mutation, we piggy-back on the `_data` member, which will be `nullptr` for disengaged optionals. Such values are valid to copy, and hence the copy constructor has to be prepared for `_data` being disengaged.

Reply:
This can be easily triggered:

```c++
mutation foo = make_mutation();
mutation bar = std::move(foo);
mutation k = foo;
```